### ISSUE TITLE
x86 32 Bit binary releases

### ISSUE THREAD

**Issue by nebulak** [rfjakob/gocryptfs#240]
gocryptfs has a really nice performance, but old netbooks only have 32 bit processors inside.
To support the use of old hardware, it would be nice if you could provide 32 bit binaries in your releases.
Is there a reason for not providing 32 bit binaries?

Similar issue: #122 

Thanks in advance

**Comment by rfjakob** [comment]
Good question!

Actually yes, there is a reason: The binaries I provide are so-called "static builds" that have no dependencies. This means they work on every Linux distro, but it also means that they cannot use the OpenSSL libraries! Instead, the static builds can only use the Go standard library.

The Go standard library is really fast if your CPU has AES-NI. But if you are on an older CPU, you usually don't have AES-NI, and the Go standard library is much slower than OpenSSL.

I myself develop on an older 64-bit CPU that does not have AES-NI, and this is what I get for self-compiled:
```
$ gocryptfs -speed
AES-GCM-256-OpenSSL 	 167.85 MB/s	(selected in auto mode)
AES-GCM-256-Go      	  49.60 MB/s	
[...]
```
And this is a static build (note that openssl shows as "N/A"):
```
$ ./gocryptfs -speed
AES-GCM-256-OpenSSL 	    N/A	
AES-GCM-256-Go      	  49.76 MB/s	(selected in auto mode)
[...]
```
In summary, if you are on an older CPU, you really should compile yourself to get good performance.

**Comment by nebulak** [comment]
Thank you for the explanation :+1: . 

The following possible solutions come to my mind:

  * Provide static built 32-bit binaries
    * Explain in readme.md that the binaries from the Releases page are static builds and that dynamic builds with OpenSSL are faster
  * Provide an install-script(similar to the current build script), which setups dependencies, builds gocryptfs and cleans the environment afterwards, if wanted by the user.

If you don't want to take the first option, I could write the install script with debian support as a starting point.

**Comment by rfjakob** [comment]
As for debian, you know you can just "apt install gocryptfs", right? https://packages.debian.org/search?keywords=gocryptfs&searchon=names&suite=all&section=all

**Comment by nebulak** [comment]
Current version in debian stretch is 1.2. So it's more than 1 year old... If that's not a problem and if the debian package is still maintained, then you're are right and you might just close this issue ;).
Thanks

**Comment by rfjakob** [comment]
Unless you need features from newer versions, you should be good with v1.2. However, I think installing [deb from buster](https://packages.debian.org/buster/gocryptfs), whis is 1.4.3, should work just fine.
